pkg/generate/app/imagestreamlookup.go

```diff
@@ -143,6 +143,13 @@ func (r ImageStreamSearcher) Search(precise bool, terms ...string) (ComponentMat
 						}
 						foundOtherTags = true
 
+						// This check after foundOtherTags = true, because in the case that
+						// we have an ImageStream with all tags hidden, we should behave as
+						// if it didn't exist at all.
```

Comment:
The comment I added in the code clearly didn't help much!  I've rewritten the comment; maybe it'll be more useful the second time around.  This corner is also covered by a test case.